tests/src/analysis/testqgsprocessing.cpp

```diff
@@ -11474,14 +11474,14 @@ void TestQgsProcessing::parameterAnnotationLayer()
   QCOMPARE( def->valueAsPythonString( QVariant::fromValue( QgsProperty::fromExpression( "\"a\"=1" ) ), context ), QStringLiteral( "QgsProperty.fromExpression('\"a\"=1')" ) );
 
   QCOMPARE( def->valueAsJsonObject( QVariant(), context ), QVariant() );
-  QCOMPARE( def->valueAsJsonObject( QStringLiteral( "main" ), context ), QVariant( QStringLiteral( "main" ) ) );
+  QCOMPARE( def->valueAsJsonObject( QStringLiteral( "main" ), context ), QVariant( context.project()->mainAnnotationLayer()->id() ) );
```

Comment:
I think this should still return "main", so that the json is compatible across different projects. Is that possible to do?

Reply:
I have changed `QgsProcessingUtils::layerToStringIdentifier()` to check for main annotation layer. Hope this is acceptable approach.